clipper_admin/clipper_admin/deployers/caffe2.py

```diff
@@ -0,0 +1,159 @@
+        for a model can be changed at any time with
+        :py:meth:`clipper.ClipperConnection.set_num_replicas`.
+    """
+
+    clipper_conn.register_application(name, input_type, default_output,
+                                      slo_micros)
+    deploy_caffe2_model(clipper_conn, name, version, input_type, inputs, func,
+                        pytorch_model, base_image, labels, registry,
+                        num_replicas)
+
+    clipper_conn.link_model_to_app(name, name)
+
+
+def deploy_caffe2_model(
+        clipper_conn,
+        name,
+        version,
+        input_type,
+        inputs,
+        func,
+        pytorch_model,
+        base_image="clipper/caffe2-container:{}".format(__version__),
+        labels=None,
+        registry=None,
+        num_replicas=1):
+    """This function deploys the prediction function with a PyTorch model. 
+    It serializes the PyTorch model in Onnx format and creates a container that loads it as a Caffe2 model. 
+    Parameters
+    ----------
+    clipper_conn : :py:meth:`clipper_admin.ClipperConnection`
+        A ``ClipperConnection`` object connected to a running Clipper cluster.
+    name : str
+        The name to be assigned to both the registered application and deployed model.
+    version : str
+        The version to assign this model. Versions must be unique on a per-model
+        basis, but may be re-used across different models.
+    input_type : str
+        The input_type to be associated with the registered app and deployed model.
+        One of "integers", "floats", "doubles", "bytes", or "strings".
+    inputs : 
+        input of func.
+    func : function
+        The prediction function. Any state associated with the function will be
+        captured via closure capture and pickled with Cloudpickle.
+    pytorch_model : pytorch model object
+        The Pytorch model to save.
+    base_image : str, optional
+        The base Docker image to build the new model image from. This
+        image should contain all code necessary to run a Clipper model
+        container RPC client.
+    labels : list(str), optional
+        A list of strings annotating the model. These are ignored by Clipper
+        and used purely for user annotations.
+    registry : str, optional
+        The Docker container registry to push the freshly built model to. Note
+        that if you are running Clipper on Kubernetes, this registry must be accesible
+        to the Kubernetes cluster in order to fetch the container from the registry.
+    num_replicas : int, optional
+        The number of replicas of the model to create. The number of replicas
+        for a model can be changed at any time with
+        :py:meth:`clipper.ClipperConnection.set_num_replicas`.
+    """
+
+    serialization_dir = save_python_function(name, func)
+
+    try:
+        torch_out = torch.onnx._export(
+            pytorch_model, inputs, "pytorch_model.onnx", export_params=True)
```

Comment:
Save this as just `"model.onnx"`, the Caffe2 model can load any ONNX model, not just ones saved from PyTorch.